tutorials/packaging/snap-a-website.md

```diff
@@ -0,0 +1,670 @@
+      - libxtst6
+      - libnss3
+      - libxss1
+      - fontconfig-config
+      - gconf2
+      - libasound2
+      - pulseaudio
+
+    after:
+      - desktop-glib-only
+
+    build: node_modules/.bin/electron-builder
+    install: |
+      mkdir $SNAPCRAFT_PART_INSTALL/app
+      mv dist/linux-unpacked/* $SNAPCRAFT_PART_INSTALL/app
+    prime:
+      - -node_modules
+
+apps:
+  castlearena:
+    command: env TMPDIR=$XDG_RUNTIME_DIR desktop-launch $SNAP/app/castlearena
+    plugs:
+      - home
+      - x11
+      - unity7
+      - browser-support
+      - network
+      - gsettings
+      - pulseaudio
+      - opengl
+```
+
+### Icon and desktop file
+
+The last bit of packaging we need is an icon and a `<app>.desktop` file so desktop environments recognize our app as such. The icon and the desktop file will be picked up by snapcraft and handled accordingly by putting them in a `snap/gui/` directory.
+
+Create this directory by running, from the root of our project:
+
+```bash
+mkdir snap/gui
+```
+
+Now, let's add an icon.
+
+#### The icon
+
+This one is a good match for our app, download this one and save it as `icon.png` in the `snap/gui/` directory.
+
+![](https://assets.ubuntu.com/v1/b2af323c-icon.png?w=256)
+
+Positive
+: An icon size of 256x256px is generally a safe bet for desktop icons to look good under most circumstances.
+
+### The `<app>.desktop` file
+
+`.desktop` files are a widely supported standard for Linux desktops. They allow the desktop shell to know about your application and display it along other apps.
+
+Create the following file and save it as `castlearena.desktop` in the `snap/gui/` directory.
+
+```bash
+[Desktop Entry]
+Type=Application
+Name=Castle Arena
+Icon=${SNAP}/meta/gui/icon.png
+Categories=Game;ArcadeGame;
+Exec=castlearena
+Terminal=false
+```
+
+Most fields are self-explanatory, but note the importance of the `Name` field. We are using "Castle Arena" here for the first time and it's how our app will be presented to users in their app list once it's installed.
+
+**We are done with packaging our app.**
+
+In the next step, we are going to run snapcraft, install our new snap and test it.
+
+## Building and testing the snap
+Duration: 1:00
+
+Now that we have created all the pieces: the app, the electron metadata (`package.json`), the snapcraft metadata (`snapcraft.yaml`) and the desktop file, it's time to run snapcraft!
+
+The command we are going to use is `snapcraft cleanbuild`: it will create an Ubuntu container using LXD, copy our project into it, download and build our part, compress the result as a snap and move the resulting snap file back into our project directory.
+
+From the root of our project (`castlearena/`), run:
+
+```bash
+snapcraft cleanbuild
+```
+
+Some details about the output you are now looking at:
+
+1. The container is created (if it's the first time you create an Ubuntu 16.04 container on this computer, an Ubuntu image is downloaded as part of the process)
```

Comment:
doesn't codelabs support proper markdown lists? These should all be '1.'

Reply:
Codelabs do, I obviously don't :D

Reply:
(fixed)

Reply:
test comment